docs/apache-airflow/security/index.rst

```diff
@@ -0,0 +1,163 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Security
+========
+
+.. toctree::
+    :maxdepth: 1
+    :glob:
+
+    *
+    secrets/index
+
+
+This document describes Airflow's security model from the perspective of
+the Airflow user. It is intended to help users understand the security
+model and make informed decisions about how to deploy and manage Airflow.
+If you would like to know, how to report security vulnerabilities and how
+security reports are handled by the security team of Airflow, head to
+`Airflow's Security Policy <https://github.com/apache/airflow/security/policy>`_.
+
+Airflow Security Model - user types
+-----------------------------------
+
+The Airflow security model involves different types of users with
+varying access and capabilities:
+
+1. **Deployment Managers**: They have the highest level of access and
+   control. They install and configure Airflow, make decisions about
+   technologies and permissions. They can potentially delete the entire
+   installation and have access to all credentials. Deployment Managers
+   can also decide to keep audits, backups and copies of information
+   outside of Airflow, which are not covered by Airflow's security
+   model.
+
+2. **DAG Authors**: They can upload, modify, and delete DAG files,
+   running code and accessing credentials. They have full access to the
+   metadata database and internal audit logs.
+
+3. **Authenticated UI users**: They have access to the UI and API. Admin
```

Comment:
Should we add an initial statement here something like "Broadly, there are three types of authenticated UI users, namely Admin users, Connection configuration users, and Operations users." and then go about explaining what they mean?

Reply:
I had it like that initially but it feels quite repetitive. I limited myself (original description was quite a bit longer and while my natural way would be similar, I think we do not loose much without this "preamble".